beetsplug/replaygain.py

```diff
@@ -81,12 +81,28 @@ def __init__(self, config, log):
         self._log = log
 
     def compute_track_gain(self, items):
+        """Computes the track gain of the given tracks, returns a list
+        of Gain objects.
+        """
+
         raise NotImplementedError()
 
     def compute_album_gain(self, album):
-        # TODO: implement album gain in terms of track gain of the
-        # individual tracks which can be used for any backend.
-        raise NotImplementedError()
+        """Computes the album gain of the given album, returns an
+        AlbumGain object.
+        """
+        # calculate track gains
+        tracks = self.compute_track_gain(album.items())
+
+        # calculate album gain
+        album_gain = sum(track.gain for track in tracks) / len(tracks)
```

Comment:
As you mentioned in the PR thread, we probably need a more "interesting" average than the plain arithmetic mean here.

Reply:
Done by now, forgot to 'resolve conservation'.
We weigh the track gains with the number of gating blocks above the threshold.